sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -70,37 +70,40 @@ export interface DeadLetterOptions {
 export { delay }
 
 // @public
-export type DeleteQueueResponse = QueueResponse;
+export type DeleteQueueResponse = EmptyResponse;
 
 // @public
-export type DeleteRuleResponse = RuleResponse;
+export type DeleteRuleResponse = EmptyResponse;
 
 // @public
-export type DeleteSubscriptionResponse = SubscriptionResponse;
+export type DeleteSubscriptionResponse = EmptyResponse;
 
 // @public
-export type DeleteTopicResponse = TopicResponse;
+export type DeleteTopicResponse = EmptyResponse;
 
 export { Delivery }
 
 // @public
-export type GetQueueResponse = QueueResponse;
+export type EmptyResponse = {
+    _response: HttpOperationResponse;
+};
+
+// @public
+export type GetQueueResponse = QueueResponse | EmptyResponse;
 
 // @public
-export type GetRuleResponse = RuleResponse;
+export type GetRuleResponse = RuleResponse | EmptyResponse;
```

Comment:
This would be a valid response for cases when we attempt a get on non-existent entities.
This API behaves differently for across entity types though. Agree that it makes sense to leave it to the service for most part (specifically fixing/addressing when and why error is thrown). 

The service does often return "Resource Conflict Occurred. Another conflicting operation may be in progress. If this is a retry for failed operation, background cleanup is still pending. Try again later." that user can better choose on how to respond to.

Reply:
> This would be a valid response for cases when we attempt a get on non-existent entities.

For GetOperations, as a user, I wouldnt expect `EmptyResponse` to be a valid response when asking for a non existing entity.

> The service does often return "Resource Conflict Occurred. Another conflicting ...

Is this treated as an error by us and bubbled up to the user appropriately?

Reply:
@ramya0820 
Are you saying that the service actually returns a 200 if you try to GET a resource that doesn't exist?

Reply:
Yes, for Queues and Topics service returns a 200. For Subscriptions and Rules, it does throw a 404 error.
Updated, at least to make experience uniform.

> Is this treated as an error by us and bubbled up to the user appropriately?

Yes, service side errors get surfaced to the user

Reply:
I would like to remove EmptyResponse from the happy path `Get*` scenarios. Having a union with EmptyResponse creates a fair amount of friction for TS users who now have to use control flow narrowing to fall into either case.

My feeling is that Queues and Topics 200 response when queue/topic is not found should be a 404 and the fact that it isn't is basically a service bug. We should probably be lifting such responses to an error. This should be easier on users.

Reply:
Agree, updated